Resources/ProtobufDumper/ProtobufDumper/ProtobufDumper.cs

```diff
@@ -0,0 +1,957 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using google.protobuf;
+using ProtoBuf;
+
+namespace ProtobufDumper
+{
+    class ProtobufDumper
+    {
+        public delegate void ProcessProtobuf( string name, string buffer );
+
+        private readonly IList<FileDescriptorProto> protobufs;
```

Comment:
Why make it an `IList`? It's private, so we can comfortably use a concrete class, and `List<T>` offers faster enumeration than `IList<T>`.